pennylane/qnodes/base.py

```diff
@@ -360,15 +434,10 @@ def _construct(self, args, kwargs):
             if self.mutable:
                 # it's ok to directly pass auxiliary arguments since the circuit is re-constructed each time
                 # (positional args must be replaced because parameter-shift differentiation requires Variables)
-                res = self.func(*arg_vars, **kwargs)
+                res = self.func(*self.arg_vars, **kwargs)
```

Comment:
How come this has been changed to an attribute?

Reply:
I don't remember. But I changed it back

Reply:
Right now my understanding would be that this is because of the caching functionality